tests/test_dynamic_resource.py

```diff
@@ -130,14 +128,6 @@ def _render(state):
 <h1>Greetings, program!</h1>
 """
 
-def test_render_renders(harness):
```

Comment:
The test immediately below does exactly that, maybe we should rename it to `test_render_renders`?